sdk/apimanagement/api-management-custom-widgets-tools/src/utils.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Key for a search param, from which editor data will be loaded from.
+ */
+export const EDITOR_DATA_KEY = "editorData";
+/**
+ * Key for a post message object, it's used to propagate changes from editor to the DevPortal. Used to prevent interference with other applications.
+ */
+export const ON_CHANGE_MESSAGE_KEY = "customInputValueChangedMSAPIM";
+/**
+ * Key for a post message object, it's used to request and send secrets - token and user id, from the DevPortal. Used to prevent interference with other applications.
+ */
+export const ASK_FOR_SECRETS_MESSAGE_KEY = "askForSecretsMSAPIM";
+
+/**
+ * Base of a values obj
+ */
+export type TValuesCommon = Record<string, unknown>;
+/**
+ * All possible runtime environments
+ */
+export type TEnvironment = "development" | "publishing" | "runtime" | "error";
+
+/** Information about the widget instance received from the Dev Portal */
+export interface TPortalData {
+  /** web content's origin (URL) of your Dev Portal */
+  origin: string;
+  /** current runtime environment */
+  environment: TEnvironment;
+  /** ID of this particular instance of the widget */
+  instanceId: string;
+}
+
+/** JSON object with all the data you'll receive from the Dev Portal */
+export interface TEditorData<TValues extends TValuesCommon> extends TPortalData {
+  /** values you've set in the admin editor window */
+  values: TValues;
+}
+
+export function getEditorDataPure<TValues extends TValuesCommon>(
+  valuesDefault: TValues,
+  urlSearchParams: URLSearchParams
+): TEditorData<TValues> {
+  try {
+    const urlEditorParams = JSON.parse(
+      decodeURIComponent(urlSearchParams.get(EDITOR_DATA_KEY) ?? "")
+    );
+
+    if (!("origin" in urlEditorParams)) {
+      console.error(
+        "Could not get 'origin' from the search params of the URL:\n" + self.location.href
+      );
+    }
+    return { ...urlEditorParams, values: { ...valuesDefault, ...urlEditorParams.values } };
+  } catch (e) {
+    console.error(
+      `Could not get '${EDITOR_DATA_KEY}' from the search params of the URL:\n` + self.location,
+      e
+    );
+    return { values: valuesDefault, environment: "error", origin: "error", instanceId: "error" };
+  }
+}
+
+/**
+ * Function to get all editor data
+ *
+ * @param valuesDefault - object with your default values to use, just import valuesDefault object from values.ts folder
+ */
+export function getEditorData<TValues extends TValuesCommon>(
+  valuesDefault: TValues
```

Comment:
Why not take another parameter called `currentLocationSearch: string` and let the consumer get it from `self.location`.

Reply:
Basically, why not just export `getEditorValuesPure` as `getEditorValues`?

Reply:
I want to simplify the API to the customers as much as possible and avoid potential issues with customers not using typescript forgetting to input self.location. This is URL of an iframe so customers don't see it at all and might get confused.